File: C++/프로그래머스/0/120849. 모음 제거/모음 제거.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

const vector<char> gather = {'a', 'e', 'i', 'o', 'u'};

string solution(string my_string) {
    string answer = my_string;
    
    for(const auto g: gather)
        answer.erase(remove(answer.begin(), answer.end(), g), 
                     answer.end());
    
    return answer;
}

File: Go/프로그래머스/0/120849. 모음 제거/모음 제거.go
func solution(my_string string) string {
    vowels := map[rune]bool{'a': true, 'e': true, 'i': true, 'o': true, 'u': true}
    
    var res []rune
    
    for _, r := range my_string {
        if !vowels[r] {
            res = append(res, r)
        }
    }
    return string(res)
}